src/Test/Hspec/Expectations.hs

```diff
@@ -88,6 +89,22 @@ with_loc(shouldBe, (Show a, Eq a) => a -> a -> Expectation)
 actual `shouldBe` expected = expectTrue ("expected: " ++ show expected ++ "\n but got: " ++ show actual) (actual == expected)
 
 -- |
+-- @actual \`shouldBeNear\` expected@ sets the expectation that @actual@ be a
+-- floating point value near @expected@. If either value is zero we check that
+-- the absolute difference is less than epsilon (1e-15) otherwise we check if
+-- the relative difference is less than epsilon.
+with_loc(shouldBeNear, (Show a, Ord a, Floating a) => a -> a -> Expectation)
+actual `shouldBeNear` expected
+    -- If they actually are exactly equal we're done.
+    | actual == expected = reportFail (actual == expected)
+    -- If either input is zero, we check if the absolute difference is less than epsilon..
+    | actual == 0 || expected == 0 = reportFail (abs (actual - expected) < 1e-15)
+    -- In all other cases we check that the relative difference is less than epsilon.
```

Comment:
source code comments are an  anti-pattern, rather done commenting make your code more expressive, e.g.

```
otherwise = reportFail (relativeDifference < epsilon)
```